src/main/java/adris/altoclef/tasks/resources/wood/CollectSticksTask.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Override
+    protected boolean shouldAvoidPickingUp(AltoClef mod) {
+        return false;
+    }
+
+    @Override
+    protected void onResourceStart(AltoClef mod) {
+        mod.getBehaviour().push();
+        mod.getBlockTracker().trackBlock(Blocks.DEAD_BUSH);
+        mod.getBlockTracker().trackBlock(ItemHelper.itemsToBlocks(ItemHelper.LOG));
+        mod.getBlockTracker().trackBlock(ItemHelper.itemsToBlocks(ItemHelper.PLANKS));
+    }
+
+    @Override
+    protected Task onResourceTick(AltoClef mod) {
+        // Calculate how much sticks can be made from planks and craft
+        int potentialStickCountFromPlanks = (mod.getItemStorage().getItemCount(ItemHelper.PLANKS)/2)*4 + mod.getItemStorage().getItemCount(Items.STICK);
+        if (potentialStickCountFromPlanks >= _targetCount) {
+            for (Item plank : ItemHelper.PLANKS) {
+                int count = mod.getItemStorage().getItemCount(plank);
+                if (count > 0) {
+                    setDebugState("Crafting sticks from planks");
+                    return new CraftInInventoryTask(new RecipeTarget(Items.STICK, (count/2)*4, generateStickRecipe(ItemHelper.PLANKS)));
+                }
+            }
+        }
+        // Do the same for logs to craft planks
+        int potentialStickCountFromLogsAndPlanks = potentialStickCountFromPlanks + mod.getItemStorage().getItemCount(ItemHelper.LOG)*8;
+        if (potentialStickCountFromLogsAndPlanks >= _targetCount) {
+            for (Item log : ItemHelper.LOG) {
+                int count = mod.getItemStorage().getItemCount(log);
+                if (count > 0) {
+                    Item plankCheck = ItemHelper.logToPlanks(log);
+                    if (plankCheck == null) {
+                        Debug.logError("Invalid/Un-convertable log: " + log + " (failed to find corresponding plank)");
+                    }
+                    int plankCount = mod.getItemStorage().getItemCount(plankCheck);
+                    int otherPlankCount = mod.getItemStorage().getItemCount(ItemHelper.PLANKS) - plankCount;
+                    int targetTotalPlanks = Math.min(count*4 + plankCount, _targetCount - otherPlankCount);
+                    setDebugState("Crafting planks from logs for sticks");
+                    return new CraftInInventoryTask(new RecipeTarget(plankCheck, targetTotalPlanks, generatePlankRecipe(ItemHelper.LOG)));
+                }
+            }
+        }
+        // Check the ground for loot
+        if (mod.getEntityTracker().itemDropped(_itemsToPickup)) {
+            return new PickupDroppedItemTask(new ItemTarget(_itemsToPickup), true);
+        }
+        // Mine wood or dead bushes
+        ArrayList<Block> blocksToMine = new ArrayList<>(Arrays.asList(ItemHelper.itemsToBlocks(ItemHelper.LOG)));
+        blocksToMine.add(Blocks.DEAD_BUSH);
+        blocksToMine.addAll(Arrays.asList(ItemHelper.itemsToBlocks(ItemHelper.PLANKS)));
```

Comment:
`blocksToMine` isn't used anywhere else, do you mean to use it on the next line instead of `ItemHelper.itemsToBlocks(_itemsToPickup)`?

Reply:
we actually don't really need a list at all if you think about Altoclef structure. The list of blocks is stored in block tracker and we only need to access one item at a time. block tracker will handle the rest.